modules/javafx.graphics/src/main/java/javafx/scene/CssStyleHelper.java

```diff
@@ -571,6 +571,140 @@ boolean pseudoClassStateChanged(PseudoClass pseudoClass) {
 
     }
 
+    // This method is a reduced version of transitionToState() method, it is added as a fix for JDK-8204568.
+    // Any modifications to the method transitionToStates() should be relatively applied here if needed.
+    // Currently this method is executed only by Labeled.fontProperty().set(), when it's font size
+    // is changed by LabeledText.
+    void recalculateRelativeSizeProperties(final Node node, Font fontForRelativeSizes) {
```

Comment:
Given the tricky nature of this fix, I would like to see a little more description in the comment explaining why it is needed and how it works. Specifically, it should say why the originally calculated value is incorrect, and indicate that a second pass is needed for certain cases.

Reply:
Updated the comment with more details.